frontend/dockerfile/builder/build.go

```diff
@@ -621,9 +608,35 @@ func Build(ctx context.Context, c client.Client) (_ *client.Result, err error) {
 	}
 
 	if scanner != nil {
-		if err := scanner(ctx, c, res, expPlatforms.IDs()); err != nil {
-			return nil, err
+		for _, p := range expPlatforms.Platforms {
+			ref, ok := res.Refs[p.ID]
+			if !ok {
+				return nil, errors.Errorf("could not find ref %s", p.ID)
+			}
+			st, err := ref.ToState()
+			if err != nil {
+				return nil, err
+			}
+
+			att, st, err := scanner(ctx, p.ID, st, nil)
```

Comment:
wdyt of controlling the extras here with defining `ARG SBOM_SCAN_STAGE=true` and global `ARG SBOM_SCAN_CONTEXT=true` (in addition to allowing it to be set in attestation attrs).